src/test/java/org/opentripplanner/transit/raptor/moduletests/D01_TimeDependentAccessTest.java

```diff
@@ -0,0 +1,241 @@
+        requestBuilder.profile(RaptorProfile.MULTI_CRITERIA)
+                .searchParams()
+                .addAccessPaths(
+                        new TimeDependentAccessEgress(
+                                STOP_2, D2m, null, hm2time(0, 18), hm2time(0, 22))
+                );
+
+        var response = raptorService.route(requestBuilder.build(), data);
+
+        assertEquals(
+                ""
+                        + "Walk 2m ~ 2 ~ BUS R2 0:20 0:35 ~ 5 ~ Walk 1m [00:18:00 00:36:00 18m, cost: 2220]\n"
+                        + "Walk 2m ~ 2 ~ BUS R3 0:25 0:40 ~ 5 ~ Walk 1m [00:20:00 00:41:00 21m, cost: 2220]",
+                pathsToString(response)
+        );
+    }
+
+    /**
+     * If AccessEgress#earliestDepartureTime returns -1, than: - openInFirstHalfIntervalTest,
+     * partiallyOpenIntervalTest don't wait at the stop - this test correctly returns an empty
+     * result.
+     *
+     * If AccessEgress#earliestDepartureTime returns `closing` time instead of `-1`, than: -
+     * openInFirstHalfIntervalTest, partiallyOpenIntervalTest correctly wait at the stop - this test
+     * incorrectly returns a trip plan before the requested departure
+     */
+    @Test
+    public void tooEarlyDeparture() {
+        requestBuilder.profile(RaptorProfile.MULTI_CRITERIA)
+                .searchParams()
+                .earliestDepartureTime(T00_30)
+                .addAccessPaths(
+                        new TimeDependentAccessEgress(STOP_2, D2m, null, T00_00, hm2time(0, 22))
+                );
+
+        var response = raptorService.route(requestBuilder.build(), data);
+
+        assertEquals(
+                ""
+                        + "Walk 2m ~ 2 ~ BUS R3 0:25 0:40 ~ 5 ~ Walk 1m [00:20:00 00:41:00 21m, cost: 2220]",
+                pathsToString(response)
+        );
+    }
+
+    private static class TimeDependentAccessEgress extends AccessEgress {
+
+        private final int opening;
+        private final int closing;
+
+        public TimeDependentAccessEgress(
+                int toFromStop,
+                int durationInSeconds,
+                State lastState,
+                int opening,
+                int closing
+        ) {
+            super(toFromStop, durationInSeconds, lastState);
+            this.opening = opening;
+            this.closing = closing;
+        }
+
+        @Override
+        public int earliestDepartureTime(int requestedDepartureTime) {
+            if (requestedDepartureTime > closing) {
+                return closing;
+            }
+            else {
+                return Math.max(requestedDepartureTime, opening);
+            }
+        }
+
+        @Override
+        public int latestArrivalTime(int requestedArrivalTime) {
+            if (requestedArrivalTime < opening) {
+                return -1;
+            }
+            else {
+                return Math.min(requestedArrivalTime, closing);
+            }
+        }
+    }
```

Comment:
I tested  this:
```
        @Override
        public int earliestDepartureTime(int requestedDepartureTime) {
            if (requestedDepartureTime < opening) {
                return opening;
            }
            else if (requestedDepartureTime > closing) {
                return opening + 24 * 3600;
            }
            return requestedDepartureTime;
        }

        @Override
        public int latestArrivalTime(int requestedArrivalTime) {
            int reqDepTime = requestedArrivalTime  - durationInSeconds();

            if (reqDepTime > closing) {
                return closing;
            }
            else if (reqDepTime < opening) {
                return closing - 24 * 3600;
            }
            return requestedArrivalTime;
        }
```
- The opening/closing hours need to  be relative  to either the access/egress  departure.
- If the access/egress schedule is every day we should not  return -1, but the next/prev day.

I did not get the expected results when running the test, so I think there are something that need to be fixed for this to work.

Hope this is helpful.

Reply:
Thanks, we've integrated the suggestions and fixed the tests so that the time usage is consistent.